lib/src/multipart_request.dart

```diff
@@ -46,18 +46,28 @@ class MultipartRequest extends BaseRequest {
   /// The list of files to upload for this request.
   final files = <MultipartFile>[];
 
-  MultipartRequest(String method, Uri url) : super(method, url);
+  /// Set a custom boundary.
+  final String _customBoundary;
```

Comment:
Some validation checks are skipped. You should run checks for the length(1 to 70 are valid) and invalid characters. Throw an exception if `_customBoundary` is not valid.

two possible ways:
1. Create getter and setter, validate inside setter.
2. Inside constructor.
I don't have strong opinion here. If you think `final` is necessary, put checks inside constructor. Otherwise, create a pair of setter and getter.

            The only mandatory parameter for the multipart  Content-Type
            is  the  boundary  parameter,  which  consists  of  1  to 70
            characters from a set of characters known to be very  robust
            through  email  gateways,  and  NOT ending with white space.
            (If a boundary appears to end with white  space,  the  white
            space  must be presumed to have been added by a gateway, and
            should  be  deleted.)   It  is  formally  specified  by  the
            following BNF:
          
            boundary := 0*69<bchars> bcharsnospace
          
            bchars := bcharsnospace / " "
          
            bcharsnospace :=    DIGIT / ALPHA / "'" / "(" / ")" / "+"  /
            "_"
                           / "," / "-" / "." / "/" / ":" / "=" / "?"